x/distribution/keeper/querier.go

```diff
@@ -158,20 +167,67 @@ func NewQueryDelegationRewardsParams(delegatorAddr sdk.AccAddress, validatorAddr
 	}
 }
 
-func queryDelegationRewards(ctx sdk.Context, path []string, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
+func queryDelegationRewards(ctx sdk.Context, _ []string, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
 	var params QueryDelegationRewardsParams
 	err := k.cdc.UnmarshalJSON(req.Data, &params)
 	if err != nil {
 		return nil, sdk.ErrUnknownRequest(sdk.AppendMsgToErr("incorrectly formatted request data", err.Error()))
 	}
+
+	// cache-wrap context as to not persist state changes during querying
 	ctx, _ = ctx.CacheContext()
+
 	val := k.stakingKeeper.Validator(ctx, params.ValidatorAddr)
 	del := k.stakingKeeper.Delegation(ctx, params.DelegatorAddr, params.ValidatorAddr)
 	endingPeriod := k.incrementValidatorPeriod(ctx, val)
 	rewards := k.calculateDelegationRewards(ctx, val, del, endingPeriod)
-	bz, err := codec.MarshalJSONIndent(k.cdc, rewards)
+
+	resp := types.DelegationRewards{
+		ValidatorAddr: params.ValidatorAddr,
+		DelegatorAddr: params.DelegatorAddr,
+		Rewards:       rewards,
+	}
+	bz, err := codec.MarshalJSONIndent(k.cdc, resp)
 	if err != nil {
 		return nil, sdk.ErrInternal(sdk.AppendMsgToErr("could not marshal result to JSON", err.Error()))
 	}
+
+	return bz, nil
+}
+
+func queryAllDelegationRewards(ctx sdk.Context, _ []string, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
+	var params QueryDelegationRewardsParams
+	err := k.cdc.UnmarshalJSON(req.Data, &params)
+	if err != nil {
+		return nil, sdk.ErrUnknownRequest(sdk.AppendMsgToErr("incorrectly formatted request data", err.Error()))
+	}
+
+	// cache-wrap context as to not persist state changes during querying
+	ctx, _ = ctx.CacheContext()
+
+	var resp types.AllDelegationRewards
```

Comment:
The desired UX may just be to sum up the rewards

Reply:
ahhh touchè

Reply:
What about having a sum as well as  `[]DelegationRewards`?

Reply:
Hmmm I think @cwgoes is right, there really isn't a need for `DelegationRewards` or `All DelegationRewards`. Having the value (or sum of values) should suffice.

Reply:
Ok thanks for the clarity here.